deepcell/image_generators/semantic.py

```diff
@@ -106,8 +106,13 @@ def __init__(self,
                              'should have rank 4. You passed an array '
                              'with shape', X.shape)
 
-        self.x = np.asarray(X, dtype=K.floatx())
-        self.y = np.asarray(y, dtype='int32')
+        if X.dtype in ['float32', 'int32', 'float64', 'int64']:
+            logging.warning('X data dtype is {}: this will increase memory use during '
+                            'preprocessing. Consider using a smaller dtype'.format(X.dtype))
+
+        if y.dtype in ['float32', 'int32', 'float64', 'int64']:
+            logging.warning('y data dtype is {}: this will increase memory use during '
+                            'preprocessing. Consider using a smaller dtype.'.format(y.dtype))
```

Comment:
Should we warn on 32? I think that's the standard size. With the other batching PR we will have far fewer memory restrictions too, so 32 might not be an obstacle.

Reply:
Agreed, makes sense